liferay-echo/liferay-portal
# LPS-178363 Cannot publish expired Web Content with Display Page whose latest version is in draft status

LPS link: [LPS-178363](https://issues.liferay.com/browse/LPS-178363)

Motivation
=======
This PR contains the solution for [LPS-178363](https://issues.liferay.com/browse/LPS-178363) which fixes the [LPP-48264](https://issues.liferay.com/browse/LPP-48264). 
Issue: If we try to publish an expired Web Content with Display Page whose latest version is in draft status, an exception will be thrown on [this line](https://github.com/liferay/liferay-portal/blob/5329fcdb461f56b6921a56252983255a7ab0076c/portal-kernel/src/com/liferay/exportimport/kernel/lar/BaseStagedModelDataHandler.java#L924) because the Staging process is publishing the draft version of the Display Page and  the draft status is not an exportable status. We noticed that the [AssetRenderer](https://github.com/liferay/liferay-portal/blob/61dcdd0157a413052accb27608505f9079fe00b0/modules/apps/journal/journal-web/src/main/java/com/liferay/journal/web/internal/asset/model/JournalArticleAssetRendererFactory.java#L108-L150) function tries to get an article with STATUS_ANY if there is no an approved version.

Proposed Solution
========
I added [two ways](https://github.com/liferay/liferay-portal/blob/51aef9a11b8a43d2fe2e4e63af9192a4ced9acde/modules/apps/journal/journal-web/src/main/java/com/liferay/journal/web/internal/asset/model/JournalArticleAssetRendererFactory.java#L125-L137) to try to get an article version with an exportable status (STATUS_EXPIRED, STATUS_SCHEDULED) before trying to get it with STATUS_ANY. So if there is any web content with exportable status, it will be got first.

Steps to Verify
========
1.Enable Local Staging
2.Create a Display Page for Basic Web Content
3.Create a Basic Web Content with this Display Page
4.Expire the Web Content
5.Edit the Web Content and save it as draft marking it as "Never Expire"
6.Publish to Live

- Expected behavior:
The Web Content with its Display Page are published successfully.
